src/actions/getZAuctionSpendAllowance.ts

```diff
@@ -1,35 +1,83 @@
-import { zAuctionConfig } from "../configuration/zAuction";
 import { TokenAllowanceParams } from "../types";
 import * as zAuction from "@zero-tech/zauction-sdk";
 import { ethers } from "ethers";
 
+import { getLogger } from "../utilities";
+
+const logger = getLogger("actions:getZAuctionSpendAllowance");
+
+type TokenAllowanceFunc<T> = (
+  account: string,
+  arg: T
+) => Promise<ethers.BigNumber>;
+
+const getAllowance = async <T>(
+  allowanceFunc: TokenAllowanceFunc<T>,
+  account: string,
+  arg: T,
+  sdk: zAuction.Instance
+): Promise<ethers.BigNumber> => {
+  let allowance = await allowanceFunc(account, arg);
+
+  if (allowance.eq("0")) {
+    allowance = await sdk.getZAuctionLegacySpendAllowance(account);
+  }
+
+  return allowance;
+};
+
 export const getZauctionSpendAllowance = async (
-  params: TokenAllowanceParams,
   account: string,
+  params: TokenAllowanceParams,
   sdk: zAuction.Instance
 ): Promise<ethers.BigNumber> => {
   let allowance: ethers.BigNumber;
+  logger.trace(`Getting allowance for user ${account}`);
```

Comment:
This function will end up printing 3 different log messages.
Could we shorten this down to 1-2?